client/src/main/kotlin/io/spine/examples/chatspn/desktop/Client.kt

```diff
@@ -0,0 +1,414 @@
+        val emailField = UserProfile.Field
+            .email()
+            .field
+            .toString()
+        val profiles = clientRequest()
+            .select(UserProfile::class.java)
+            .where(QueryFilter.eq(EntityColumn(emailField), email.toEmail()))
+            .run()
+        if (profiles.isEmpty()) {
+            return null
+        }
+        return profiles[0]
+    }
+
+    /**
+     * Creates a new personal chat.
+     */
+    public fun createPersonalChat(user: UserId) {
+        val command = createPersonalChatCommand(user, authenticatedUser!!.id)
+        clientRequest()
+            .command(command)
+            .postAndForget()
+    }
+
+    /**
+     * Sends message to the chat.
+     */
+    public fun sendMessage(chat: ChatId, content: String) {
+        val command = sendMessageCommand(chat, authenticatedUser!!.id, content)
+        clientRequest()
+            .command(command)
+            .postAndForget()
+    }
+
+    /**
+     * Returns chats that `authenticatedUser` is a member of.
+     */
+    public fun readChats(): List<ChatPreview> {
+        val chats = clientRequest()
+            .select(UserChats::class.java)
+            .byId(authenticatedUser!!.id)
+            .run()[0]
+        return chats.chatList
+    }
+
+    /**
+     * Subscribes `action` on `authenticatedUser`'s chats changes.
+     */
+    public fun subscribeOnChats(action: (state: UserChats) -> Unit) {
+        val subscription = clientRequest()
+            .subscribeTo(UserChats::class.java)
+            .byId(authenticatedUser!!.id)
+            .observe(action)
+            .post()
+        userChatsSubscriptions.add(subscription)
+    }
+
+    /**
+     * Cancel all subscriptions on `authenticatedUser`'s chats changes.
+     */
+    public fun cancelChatsSubscriptions() {
+        userChatsSubscriptions.forEach { subscription ->
+            client.subscriptions().cancel(subscription)
+        }
+        userChatsSubscriptions.clear()
+    }
+
+    /**
+     * Returns messages from the provided chat.
+     */
+    public fun readMessages(chat: ChatId): List<MessageView> {
+        val whenPostedField = MessageView.Field
+            .whenPosted()
+            .field
+            .toString()
+        val messages = clientRequest()
+            .select(MessageView::class.java)
+            .where(chat.queryFilter())
+            .orderBy(EntityColumn(whenPostedField), OrderBy.Direction.ASCENDING)
+            .run()
+        return messages
+    }
+
+    /**
+     * Subscribes actions on messages in the chat.
+     *
+     * @param chat id of the chat to subscribe on messages in
+     * @param updateAction an action that will be triggered when a new chat message is posted,
+     *                     or when an existing message is updated
+     * @param deleteAction an action that will be triggered when a message is deleted
+     */
+    public fun subscribeOnMessages(
```

Comment:
`subscribeToMessages`

I would even say `observeMessages` (and `stopObservingMessages` below).